### PR TITLE
Changed existing users UserNames to random strings

### LINKED ISSUE #353: Change existing users UserNames
All previous user has UserName = Email. We should find the solution to let them set UserName (once)

Maybe add some flag in Db `AllowChangeUserName`?

### PR DESCRIPTION
**Pull request by smiechuskce**
Added a notification about a possibility to change user's username if his user name is same as the email. Added a script (a migration) which changes all user names to random 12 chars strings.

Also, added a test for a possiblity to update user name to the one that exists. **This test doesn't pass. So, please help with that. ASP.NET Identity CreateAsync() method always returns null despite of being configured with a proper mock to return a specific value. I have no idea how to encourage it to work properly...**

## Where should the reviewer start?
Update database, run the application, run the tests.

## Motivation and context

closes #353 

## How has this been tested?
Update database, check for changes in UserName & NormalizedUserName.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by WTobor** [comment]
@Mejtika could you help with that?

**Comment by Mejtika** [comment]
Hi @smiechuskce! :) I've dug into this code trying to find a solution, but with going deeper into userManager internals I was losing myself more and more. We also currently have another PR #397 where we're trying to test the account controller, but we're having difficulties doing this because mocking some dependencies is too complex. I'm thinking about writing tests to controllers, that will use WebApplicationFactory with our Startup, which we will modify according to our needs. I think that should help us with complex mocking. What do you think about that?

**Comment by smiechuskce** [comment]
> Hi @smiechuskce! :) I've dug into this code trying to find a solution, but with going deeper into userManager internals I was losing myself more and more. We also currently have another PR #397 where we're trying to test the account controller, but we're having difficulties doing this because mocking some dependencies is too complex. I'm thinking about writing tests to controllers, that will use WebApplicationFactory with our Startup, which we will modify according to our needs. I think that should help us with complex mocking. What do you think about that?

Hi, many thanks for help ;) I've got similar troubles mocking all the needed dependecies - it's hard to me to specify which one of them causes CreateAsync to return null.

Writing tests using WebApplicationFactory looks OK to me, because you'd send a request to the server using, for example HttpClient and verify the response. But I think it doesn't solve the problem we've got here. In fact, we need to test if the CreateAsync produces IdentityResult with Errors indicating we got a duplicate user here. And then the exception would be thrown (or not - it's actually not necessary). We still need to use a proper implementation of UserValidator, UserStore and UserManager.

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Manage/Index.cshtml [hunk 1]
```diff
@@ -14,6 +16,11 @@
 <div class="row">
     <div class="col-md-12">
         <form method="post" class="contact_form">
+            @if (Model.Email == Model.Username)
+            {
+                <div class="alert alert-info" role="alert">Możesz teraz ustawić sobie nazwę użytkownika! Nazwa ta będzie widoczna publicznie w tabeli wyników.</div>
```
Comment (WTobor) [inline src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Manage/Index.cshtml:21]: please move it to resources

Comment (smiechuskce) [reply src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Manage/Index.cshtml]: done

**Review by WTobor — APPROVED** [review]
